RiotSwiftUI/Modules/UserSessions/DeviceType/DeviceType+Element.swift

```diff
@@ -0,0 +1,58 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import SwiftUI
+
+extension DeviceType {
+    
+    var image: Image {
+        
+        let image: Image
+        
+        switch self {
+        case .desktop:
+            image = Image(Asset.Images.deviceTypeDesktop.name)
+        case .web:
+            image = Image(Asset.Images.deviceTypeWeb.name)
+        case .mobile:
+            image = Image(Asset.Images.deviceTypeMobile.name)
+        case .unknown:
+            image = Image(Asset.Images.deviceTypeUnknown.name)
+        }
+        
+        return image
+    }
+    
+    var name: String {
+        let name: String
+        
+        let appName = AppInfo.current.displayName
```

Comment:
Will this not show the _current_ app's bundle display name for all devices?

Reply:
Yes atm the application short name `Element` is taken from the bundle display name in the app. This is probably something we set in other place like `BuildSettings` or specific translation file in the future.
The parsing of client name is not yet implemented. So there is chances to change the client name generation in the future also.